src/components/Canary/minimal.js

```diff
@@ -20,23 +21,39 @@ const MinimalCanaryFC = ({
   );
 
   useEffect(() => {
-    history.listen(({ location }) => {
+    const unlisten = history.listen(({ location }) => {
       setSearchParams(getParamsFromURL(location.search));
```

Comment:
It's existing code, but since changes are related, consider replacing the logic with react-router useSearchParams?